nftables_test.go

```diff
@@ -2457,6 +2457,136 @@ func TestCreateUseAnonymousSet(t *testing.T) {
 	}
 }
 
+func TestCappedErrMsgOnObj(t *testing.T) {
+	c, newNS := openSystemNFTConn(t)
+	c, err := nftables.New(nftables.WithNetNSFd(int(newNS)), nftables.AsLasting())
+	if err != nil {
+		t.Fatalf("nftables.New() failed: %v", err)
+	}
+	defer cleanupSystemNFTConn(t, newNS)
+	c.FlushRuleset()
+	defer c.FlushRuleset()
+
+	filter := c.AddTable(&nftables.Table{
+		Family: nftables.TableFamilyIPv4,
+		Name:   "filter",
+	})
+	if err := c.Flush(); err != nil {
+		t.Errorf("failed adding table: %v", err)
+	}
+
+	const counterLen = 500
+	counters := make([]nftables.Obj, counterLen)
+	for i := 0; i < counterLen; i++ {
+		counters[i] = &nftables.CounterObj{
+			Table:   filter,
+			Name:    fmt.Sprintf("ctr%d", i),
+			Bytes:   uint64(-1 * i),
+			Packets: uint64(-1 * i),
+		}
+		c.AddObj(counters[i])
+	}
+
+	if err := c.Flush(); err != nil {
+		if got, want := fmt.Sprint(err), "no buffer space available"; !strings.Contains(got, want) {
+			t.Errorf("expected error %s, got %v", want, err)
+		}
+	}
+
+	// due to large number of counters
+	// netlink will return the "no error" message
```

Comment:
Do you happen to know the corresponding kernel code? Might be good to add a link for folks who want to understand this behavior in detail :)

Reply:
I have rephrased and added additional information regarding this test case. It is not about the kernel code, it is related to the netlink go library which tries to peek for data that is too large to fit the socket read buffer. This results in not receiving the "no error" ack messages when expected and some of those messages end up as a response to the `GetObj` call done afterwards.